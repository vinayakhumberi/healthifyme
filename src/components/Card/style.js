import styled from 'styled-components';
import { rem } from '../../utils/style.util';

export const Container = styled.div`
  width: ${props =>
    typeof props.width === 'string' ? props.width : rem(props.width)};
  height: ${props =>
    typeof props.height === 'string' ? props.height : rem(props.height)};
  padding-right: ${props =>
    typeof props.paddingRight === 'string'
      ? props.paddingRight
      : rem(props.paddingRight)};
  padding-left: ${props =>
    typeof props.paddingLeft === 'string'
      ? props.paddingLeft
      : rem(props.paddingLeft)};
  padding-bottom: ${props =>
    typeof props.paddingBottom === 'string'
      ? props.paddingBottom
      : rem(props.paddingBottom)};
  padding-top: ${props =>
    typeof props.paddingTop === 'string'
      ? props.paddingTop
      : rem(props.paddingTop)};
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  text-align: ${props => props.textAlign};
`;