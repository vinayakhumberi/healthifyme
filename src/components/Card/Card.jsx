import React from "react";
import PropTypes from "prop-types";
import { Container } from "./style";

const Card = function (props) {
  return (
    <Container
      data-test="card-container"
      {...props}
    >
     {props.children}
    </Container>
  );
};
Card.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  paddingRight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingLeft: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingTop: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingBottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  textAlign: PropTypes.string,
};

Card.defaultProps = {
  width: "auto",
  height: "auto",
  paddingRight: "inherit",
  paddingLeft: "inherit",
  paddingTop: "inherit",
  paddingBottom: "inherit",
  textAlign: "initial",
};

export default Card;
