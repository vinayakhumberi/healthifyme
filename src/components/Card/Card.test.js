import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('<Card />', () => {
  let wrapper;
  const text = "This is a test card";
  beforeEach(() => wrapper = shallow(<Card>{text}</Card>));

  it('Should Render a Card Container', () => {
    expect(wrapper.find('[data-test="card-container"]').length).toEqual(1);
  });
  it('Should Render a Card Container', () => {
    expect(wrapper.text()).toEqual(text);
  });
});



describe('<Card {...props} />', () => {
  let wrapper;
  const text = "This is a test card";
  let props = {
    width: "auto",
    height: "auto",
    paddingRight: "inherit",
    paddingLeft: "inherit",
    paddingTop: "inherit",
    paddingBottom: "inherit",
    textAlign: "initial",
  };
  beforeEach(() => wrapper = shallow(<Card {...props} >{text}</Card>));

  it('Should Render a Card Container', () => {
    expect(wrapper.find('[data-test="card-container"]').length).toEqual(1);
  });
  it('Should Render a Card Container', () => {
    expect(wrapper.text()).toEqual(text);
  });

  it('Should pass backgroundColor prop to Button as auto', () => {
    expect(wrapper.find('[data-test="card-container"]').get(0).props.width).toEqual(props.width);
  });
  it('Should pass backgroundColor prop to Button as auto', () => {
    expect(wrapper.find('[data-test="card-container"]').get(0).props.height).toEqual(props.height);
  });
  it('Should pass backgroundColor prop to Button as inherit', () => {
    expect(wrapper.find('[data-test="card-container"]').get(0).props.paddingRight).toEqual(props.paddingRight);
  });
  it('Should pass backgroundColor prop to Button as inherit', () => {
    expect(wrapper.find('[data-test="card-container"]').get(0).props.paddingLeft).toEqual(props.paddingLeft);
  });
  it('Should pass backgroundColor prop to Button as inherit', () => {
    expect(wrapper.find('[data-test="card-container"]').get(0).props.paddingTop).toEqual(props.paddingTop);
  });
  it('Should pass backgroundColor prop to Button as inherit', () => {
    expect(wrapper.find('[data-test="card-container"]').get(0).props.paddingBottom).toEqual(props.paddingBottom);
  });
  it('Should pass backgroundColor prop to Button as initial', () => {
    expect(wrapper.find('[data-test="card-container"]').get(0).props.textAlign).toEqual(props.textAlign);
  });
});