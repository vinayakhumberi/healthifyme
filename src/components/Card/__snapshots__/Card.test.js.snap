// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card {...props} /> renders correctly 1`] = `ShallowWrapper {}`;
